packages/checkout-storefront/src/sections/PaymentSection/AdyenDropIn/useAdyenDropin.ts

```diff
@@ -0,0 +1,321 @@
+            });
+          }
+        },
+      }),
+      [
+        adyenCheckoutSubmitParams?.component,
+        commonErrorMessages.somethingWentWrong,
+        handlePaymentResult,
+        showCustomErrors,
+        transactionInitialize,
+      ]
+    )
+  );
+
+  const onTransactionProccess = useSubmit<
+    TransactionProcessMutationVariables,
+    typeof transactionProccess
+  >(
+    useMemo(
+      () => ({
+        onSubmit: transactionProccess,
+        onError: () => {
+          // will tell the processing screen to disappear
+          setIsProcessingPayment(false);
+          // we don't do these at onFinished since redirect will happen first
+          clearQueryParams("transaction");
+          setCurrentTransactionId(null);
+
+          showCustomErrors([{ message: commonErrorMessages.somethingWentWrong }]);
+          adyenCheckoutSubmitParams?.component.setStatus("ready");
+        },
+        extractCustomErrors: (result) => result?.data?.transactionProcess?.data?.errors,
+        onSuccess: ({ data }) => {
+          if (!data?.data) {
+            showCustomErrors([{ message: commonErrorMessages.somethingWentWrong }]);
+            return;
+          }
+
+          const {
+            transaction,
+            data: { paymentDetailsResponse },
+          } = data;
+
+          // we don't do these at onFinished since redirect will happen first
+          clearQueryParams("transaction");
+          setCurrentTransactionId(null);
+
+          handlePaymentResult({
+            paymentResponse: paymentDetailsResponse,
+            transaction,
+          });
+        },
+      }),
+      [
+        adyenCheckoutSubmitParams?.component,
+        commonErrorMessages.somethingWentWrong,
+        handlePaymentResult,
+        setIsProcessingPayment,
+        showCustomErrors,
+        transactionProccess,
+      ]
+    )
+  );
+
+  const onSubmitInitialize: AdyenCheckoutInstanceOnSubmit = useEvent(async (state, component) => {
+    component.setStatus("loading");
+    setAdyenCheckoutSubmitParams({ state, component });
+    validateAllForms(authenticated);
+    setSubmitInProgress(true);
+  });
+
+  useEffect(() => {
+    const validating = anyFormsValidating(validationState);
+    const allFormsValid = areAllFormsValid(validationState);
+
+    // any of the conditions below - do nothing
+    if (!submitInProgress || validating || anyRequestsInProgress || !adyenCheckoutSubmitParams) {
+      return;
+    }
+
+    // there was en error either in some other request or form validation
+    // - stop the submission altogether
+    if (!finishedApiChangesWithNoError || !allFormsValid) {
+      adyenCheckoutSubmitParams?.component.setStatus("ready");
+      setSubmitInProgress(false);
+      return;
+    }
+
+    // submit in progress only means that submit has been initialized
+    // we want to disable it here so it's not run again until the user
+    // initializes manually
+    setSubmitInProgress(false);
```

Comment:
Shouldn't this be set to true?
```suggestion
    setSubmitInProgress(true);
```

Reply:
Definitely should be `false`. Seems like the comment I added here isn't clear enough tho... 🤔 To explain: we want to disable it here because the `true` set earlier has served its purpose - it enabled the useEffect to reach this place where it's either gonna run `onTransactionProcess` or `onTransactionInitialize`. If we left it as `true`, the next time the useEffect runs (which is a lot, because this useEffect has a lot of different deps) it could run these mutations again, which we don't want, because we want them to only run once the user presses `submit`. So the user presses submit -> submit in progress: true + validate all forms -> useEffect catches these changes, awaits for forms validation and other requests to finish -> if everything is cool then sets submit in progress: false + runs the proper mutation. How could the comment above be phrased better so it'd express this properly?